tests/core/engine_adapter/test_integration.py

```diff
@@ -376,6 +395,116 @@ def non_temp_tables(self) -> t.List[str]:
         return [x for x in self.tables if not x.startswith("__temp") and not x.startswith("temp")]
 
 
+class ProjectCreator(PydanticModel):
+    """
+    Create test-scoped projects in the tmpdir of the currently executing test, under a schema specific to the currently executing test
+
+    The intention is to use these to set up bare minimum SQLMesh projects that can be used to end-to-end test a single thing
+    """
+
+    SEEDS_DIR: str = "seeds"
+    MODELS_DIR: str = "models"
+
+    output_path: pathlib.Path
+    engine_adapter: EngineAdapter
+    gateway: str
+    schema_name: str
+    _context: t.Optional[Context] = PrivateAttr(default=None)
+
+    def add_seed(self, model_name: str, columns: t.Dict[str, str], rows: t.List[t.Dict[str, str]]):
```

Comment:
check out core/dialect.py::pandas_to_sql

or just use create_view in the engine_adapter, that accepts a pandas dataframe

Reply:
So the goal was to produce a project from a user perspective, based on the files in the filesystem that a user would create.

A user doesn't create a seed as a pandas DataFrame (unless theyre using Python models I guess), they create a .csv file in the `seeds/` directory and expose it in a SQL model using `kind SEED`.

However, i've implemented the `pandas_to_sql` version because in this case the seed data is just a vehicle to expose an issue in the INCREMENTAL_BY_UNIQUE_KEY model

Reply:
I ended up removing this entirely in favour of adding the core methods to `TestContext` instead